docs/_plugins/search_index.rb

```diff
@@ -481,34 +480,77 @@ def is_latest?(group, model)
       model[:download_link] = models_json[model[:url]][:download_link]
 
       if client
-        if force_reindex || uniq_for_indexing.include?(uniq)
-          if is_latest?(name_language_editions_sparkversion_to_models_mapping, model)
-            id = model.delete(:id)
-            bulk_indexer.index(id, model)
-          end
+        if is_latest?(name_language_editions_sparkversion_to_models_mapping, model)
+          id = model.delete(:id)
+          model.delete(:uniq_key)
+          bulk_indexer.index(id, model)
         end
       end
     end
   end
   bulk_indexer.execute
 
-  if client
-    # Also delete models whose  name, language, edition, spark version were modified
+  if client and not is_incremental
```

Comment:
We have to delete models from the Elasticsearch index in the incremental builds as well.

Reply:
Removed models will appear via this hooks 
Jekyll::Hooks.register :clean, :on_obsolete do |files|
I have implemented deleted models here for incremental build